tests/functional/test_workspaces.py

```diff
@@ -0,0 +1,151 @@
+        except exceptions.HTTPError as e:
+            if e.response.status_code != 404:
+                raise
+        try:
+            self.buckets.delete('in.c-py-test-buckets', force=True)
+        except exceptions.HTTPError as e:
+            if e.response.status_code != 404:
+                raise
+        # https://github.com/boto/boto3/issues/454
+        warnings.simplefilter("ignore", ResourceWarning)
+
+    def tearDown(self):
+        try:
+            if hasattr(self, 'workspace_id'):
+                self.workspaces.delete(self.workspace_id)
+        except exceptions.HTTPError as e:
+            if e.response.status_code != 404:
+                raise
+        try:
+            self.buckets.delete('in.c-py-test-tables', force=True)
+        except exceptions.HTTPError as e:
+            if e.response.status_code != 404:
+                raise
+
+    def test_create_workspace(self):
+        workspace = self.workspaces.create()
+        self.workspace_id = workspace['id']
+        with self.subTest():
+            self.assertTrue('id' in workspace)
+        with self.subTest():
+            self.assertTrue('type' in workspace)
+            self.assertTrue(workspace['type'] in ['table', 'file'])
+        with self.subTest():
+            self.assertTrue('name' in workspace)
+        with self.subTest():
+            self.assertTrue('component' in workspace)
+        with self.subTest():
+            self.assertTrue('configurationId' in workspace)
+        with self.subTest():
+            self.assertTrue('created' in workspace)
+        with self.subTest():
+            self.assertTrue('connection' in workspace)
+        with self.subTest():
+            self.assertTrue('backend' in workspace['connection'])
+        with self.subTest():
+            self.assertTrue('creatorToken' in workspace)
+
+    def test_load_tables_to_workspace(self):
+        bucket_id = self.buckets.create('py-test-tables')['id']
+        table1_id = self.__create_table(bucket_id, 'test-table-1', {'col1': 'ping', 'col2': 'pong'})
+        table2_id = self.__create_table(bucket_id, 'test-table-2', {'col1': 'king', 'col2': 'kong'})
+        workspace = self.workspaces.create()
+        self.workspace_id = workspace['id']
+        job = self.workspaces.load_tables(
+            workspace['id'],
+            {table1_id: 'destination_1', table2_id: 'destination_2'}
+        )
+        self.jobs.block_until_completed(job['id'])
+
+        conn = self.__get_snowflake_conenction(workspace['connection'])
+        cursor = conn.cursor()
+        cursor.execute('SELECT * FROM "destination_1"')
+        self.assertEqual(('ping', 'pong'), cursor.fetchone())
+        cursor.execute('SELECT * FROM "destination_2"')
+        self.assertEqual(('king', 'kong'), cursor.fetchone())
+        conn.close()
+
+    # test load files into an abs workspace
+    def test_load_files_from_workspace(self):
+        if (os.getenv('SKIP_ABS_TEST')):
```

Comment:
should be part of mock tests (if needed at all)

Reply:
@odinuv not sure what you mean by this, that loading files to the file workspace shouldn't be functionally tested?

Reply:
ah, if you meant the invalid workspace test throwing the exception then that makes sense, yeah that could go in the mock tests